vanguard.go

```diff
@@ -427,3 +463,19 @@ func descKind(desc protoreflect.Descriptor) string {
 		return fmt.Sprintf("%T", desc)
 	}
 }
+
+func withDefaultResolver(service protoreflect.ServiceDescriptor) ServiceOption {
```

Comment:
I don't think this should be an option if it's a non public method. Options should be user facing. Would it be possible to create the default resolver without an option.

Reply:
Sure.

Reply:
done